core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -249,6 +258,30 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       }
     }
 
+    /**
+     * Mark the given executor as disabled; disabled executors are as good as lost, except their
+     * tasks are not yet marked as failed.
```

Comment:
can you expand on this to explain why we need this intermediate step in the middle?